crates/matrix-sdk-crypto/src/verification/requests.rs

```diff
@@ -78,9 +78,9 @@ pub enum VerificationRequestState {
         /// The verification methods supported by the sender.
         their_methods: Vec<VerificationMethod>,
 
-        /// The device ID of the device that responded to the verification
+        /// The device data of the device that responded to the verification
         /// request.
-        other_device_id: OwnedDeviceId,
+        other_device_data: DeviceData,
```

Comment:
@poljar We initially agreed to pass this `DeviceData` down into the `Transitioned` state but I couldn't identify were that would be useful 🤔

Reply:
Well that's because, when we go from the `Requested` state into the `Ready` state you throw away the `DeviceData` and the `Ready` state again only remembers the device ID.

You should propagate the `DeviceData` to the `Ready` state here:

https://github.com/matrix-org/matrix-rust-sdk/blob/1d4b34f7c92d389d9ac5ed59377822603848c764/crates/matrix-sdk-crypto/src/verification/requests.rs#L915-L916

And then further to the `Transitioned` state, this would allow you to get rid of the second `get_device_call()` we have:

https://github.com/matrix-org/matrix-rust-sdk/blob/1d4b34f7c92d389d9ac5ed59377822603848c764/crates/matrix-sdk-crypto/src/verification/requests.rs#L1354-L1362

Reply:
Nice, thank you for the very clear explanation, it should be good to go now! I've rebased the whole PR to help app developers but only the last 3 commits are new.